src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1372 @@
+
+        loss = None
+        if labels is not None:
+            loss_fct = CrossEntropyLoss(ignore_index=-100)
+            loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
+
+        if not return_dict:
+            output = (lm_logits,) + decoder_outputs[1:] + encoder_outputs
+            return ((loss,) + output) if loss is not None else output
+
+        return Seq2SeqLMOutput(
+            loss=loss,
+            logits=lm_logits,
+            past_key_values=decoder_outputs.past_key_values,
+            decoder_hidden_states=decoder_outputs.hidden_states,
+            decoder_attentions=decoder_outputs.attentions,
+            cross_attentions=decoder_outputs.cross_attentions,
+            encoder_last_hidden_state=encoder_outputs.last_hidden_state,
+            encoder_hidden_states=encoder_outputs.hidden_states,
+            encoder_attentions=encoder_outputs.attentions,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        input_features: BatchFeature,
+        inputs_embeds=None,
+        composer="composer1",
+        n_bars: int = 2,
+        max_length: int = 256,
+        inputs: Optional[torch.Tensor] = None,
+        generation_config=None,
+        **kwargs,
+    ):
+        """
+        Generates sequences of token ids for models with a language modeling head.
+
+        <Tip warning={true}>
+
+        Most generation-controlling parameters are set in `generation_config` which, if not passed, will be set to the
+        model's default generation configuration. You can override any `generation_config` by passing the corresponding
+        parameters to generate(), e.g. `.generate(inputs, num_beams=4, do_sample=True)`. For an overview of generation
+        strategies and code examples, check out the [following guide](./generation_strategies).
+
+        </Tip>
+
+        Parameters:
+            input_features (`BatchFeature`):
+                `input_features` returned by `Pop2PianoFeatureExtractor.__call__`
+            inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
+                Optionally, instead of passing `input_features` you can choose to directly pass an embedded
+                representation. It is advised to use `input_features` and not `inputs_embeds` since `inputs_embeds`
+                requires values which are passed through `LogMelSpectrogram` and `ConcatEmbeddingToMel`.
+            composer (`str`, *optional*, defaults to `"composer1"`):
+                This value is passed to `ConcatEmbeddingToMel` to generate different embeddings for each `"composer"`.
+            n_bars (`int`, *optional*, defaults to 2);
+                Helps to determine the `max_length` if `max_length` is not given.
+            max_length (`int`, *optional*):
+                Number of tokens to be generated.
+            inputs (`torch.Tensor` of varying shape depending on the modality, *optional*):
```

Comment:
Not really sure what the purpose of `inputs` is? A prompt for generation? How does this work for Pop2Piano?